From: Bill MacAllister <bill@ca-zephyr.org>
Date: Fri, 15 May 2020 18:32:56 -0700
Subject: Fix help display and syslog output of kafs-dns

This patch modifies the help display and syslog output of the kafs-dns
binary to correctly identify the binary generating the output.
---
 src/dns_main.c | 11 ++++++-----
 1 file changed, 6 insertions(+), 5 deletions(-)

diff --git a/src/dns_main.c b/src/dns_main.c
index 2c1e5a3..c50bf18 100644
--- a/src/dns_main.c
+++ b/src/dns_main.c
@@ -52,6 +52,7 @@
 
 static const char *DNS_PARSE_VERSION = "2.0";
 static const char prog[] = "dns_afsdb";
+static const char prog_bin[] = "kafs-dns";
 static const char key_type[] = "dns_resolver";
 static const char afsdb_query_type[] = "afsdb:";
 static key_serial_t key;
@@ -134,11 +135,11 @@ void usage(void)
 {
 	if (isatty(2)) {
 		fprintf(stderr,	"Usage: %s [OPTION]... <key_serial>\n",
-			prog);
+			prog_bin);
 		fprintf(stderr,	"       %s -D [OPTION]... <desc> <calloutinfo>\n",
-			prog);
+			prog_bin);
 		fprintf(stderr,	"       %s -V\n",
-			prog);
+			prog_bin);
 
 		fprintf(stderr,	"\n");
 		fprintf(stderr,	"Where [OPTION].. is a combination of one or more of:\n");
@@ -151,7 +152,7 @@ void usage(void)
 		fprintf(stderr,	"\t-o <dumpfile>\n");
 		fprintf(stderr,	"\t-v\n");
 	} else {
-		verbose("Usage: %s [-vv] <key_serial>", prog);
+		verbose("Usage: %s [-vv] <key_serial>", prog_bin);
 	}
 	exit(2);
 }
@@ -233,7 +234,7 @@ int main(int argc, char *argv[])
 	if (argc > 1 && strcmp(argv[1], "--help") == 0)
 		usage();
 
-	openlog(prog, 0, LOG_DAEMON);
+	openlog(prog_bin, 0, LOG_DAEMON);
 
 	while ((ret = getopt_long(argc, argv, "Dvc:N:o:V:", long_options, NULL)) != -1) {
 		switch (ret) {
